src/modinfo.h

```diff
@@ -130,415 +117,556 @@ class ModInfo : public QObject, public MOBase::IModInterface
+  // of them in a single place.
 
   /**
-   * @brief test if there is a newer version of the mod
-   *
-   * test if there is a newer version of the mod. This does NOT cause
-   * information to be retrieved from the nexus, it will only test version information already
-   * available locally. Use checkAllForUpdate() to update this version information
-   *
-   * @return true if there is a newer version
-   **/
-  virtual bool updateAvailable() const = 0;
+   * @return the name of the mod.
+   */
+  virtual QString name() const = 0;
 
   /**
-   * @return true if the update currently available is ignored
+   * @return the absolute path to the mod to be used in file system operations.
    */
-  virtual bool updateIgnored() const = 0;
+  virtual QString absolutePath() const = 0;
 
   /**
-   * @brief test if the "newest" version of the mod is older than the installed version
-   *
-   * test if there is a newer version of the mod. This does NOT cause
-   * information to be retrieved from the nexus, it will only test version information already
-   * available locally. Use checkAllForUpdate() to update this version information
-   *
-   * @return true if the newest version is older than the installed one
-   **/
-  virtual bool downgradeAvailable() const = 0;
+   * @return the comments for this mod, if any.
+   */
+  virtual QString comments() const = 0;
 
   /**
-   * @brief request an update of nexus description for this mod.
-   *
-   * This requests mod information from the nexus. This is an asynchronous request,
-   * so there is no immediate effect of this call.
-   * Right now, Mod Organizer interprets the "newest version" and "description" from the
-   * response, though the description is only stored in memory
-   *
-   **/
-  virtual bool updateNXMInfo() = 0;
+   * @return the notes for this mod, if any.
+   */
+  virtual QString notes() const = 0;
 
   /**
-   * @brief assign or unassign the specified category
+   * @brief Retrieve the short name of the game associated with this mod. This may differ
+   *     from the current game plugin (e.g. you can install a Skyrim LE game in a SSE
+   *     installation).
    *
-   * Every mod can have an arbitrary number of categories assigned to it
+   * @return the name of the game associated with this mod.
+   */
+  virtual QString gameName() const = 0;
+
+  /**
+   * @return the name of the repository from which this mod was installed.
+   */
+  virtual QString repository() const override { return ""; }
+
+  /**
+   * @return the ID of this mod on the repository.
    *
-   * @param categoryID id of the category to set
-   * @param active determines wheter the category is assigned or unassigned
-   * @note this function does not test whether categoryID actually identifies a valid category
-   **/
-  virtual void setCategory(int categoryID, bool active) = 0;
+   * @note For Nexus, this is the Nexus ID of the mod.
+   */
+  virtual int modId() const = 0;
```

Comment:
I would use `nexusId()` . This is used in the code to determine whether a mod is nexus stuff or not. It shouldn't be used for other repositories. The setter is setNexusId() and I feel modId can be confused to be some internal id for the mod, like the mod Index. nexusId is really clear on the other hand.

Reply:
If/when we add integration for other mod repositories, we can rename the function then. It'll probably need changing to a pair of repository name and ID when that happens anyway.

Reply:
There is already a `repository()` function that returns the repository. 

I'll switch this back to `nexusId()` when I get back home.